src/bentoml/_internal/io_descriptors/json.py

```diff
@@ -180,6 +229,7 @@ def __init__(
         self,
         *,
         pydantic_model: t.Type[pydantic.BaseModel] | None = None,
+        typeddict: t.Type[t.TypedDict] | None = None,
```

Comment:
Another thing is is typeddict and pydantic_model should be mutually exclusive.

Reply:
On top of the `__init__` , I'll add the following logic
~~~python
def __init__(...):
    if pydantic_model is not None and typeddict is not None:
        raise BadInput("typeddict and pydantic_model are mutually exclusive. use only typeddict or pydantic_model")
~~~